app/controllers/concerns/authentication_and_sso_concerns.rb

```diff
@@ -21,6 +21,15 @@ def render_unauthorized
     raise Common::Exceptions::Unauthorized
   end
 
+  def validate_authn_context
+    authn = params[:authn]
+    raise Common::Exceptions::ParameterMissing, 'authn' if authn.blank?
+    raise Common::Exceptions::InvalidFieldValue.new('authn', authn) unless
+      SAML::User::AUTHN_CONTEXTS.keys.include?(authn)
```

Comment:
Nit: Using multiple lines for an inline conditional feels weird. I think I'd break it out to the block syntax or perhaps invert the logic via `if ...exclude?` to keep it on one line.